razorpay/client.py

```diff
@@ -84,8 +84,20 @@ def _update_user_agent_header(self, options):
     def _get_version(self):
         version = ""
         try: # nosemgrep : gitlab.bandit.B110
-            version = pkg_resources.require("razorpay")[0].version
-        except DistributionNotFound:  # pragma: no cover
+            # Try importlib.metadata first (modern approach)
+            try:
+                import importlib.metadata
+                from importlib.metadata import PackageNotFoundError
+                version = importlib.metadata.version("razorpay")
+            except ImportError:
+                # Fall back to pkg_resources
+                import pkg_resources
+                from pkg_resources import DistributionNotFound
+                version = pkg_resources.require("razorpay")[0].version
+        except (PackageNotFoundError, DistributionNotFound, NameError):  # pragma: no cover
+            # PackageNotFoundError: importlib.metadata couldn't find the package
+            # DistributionNotFound: pkg_resources couldn't find the package  
+            # NameError: in case the exception classes aren't defined due to import issues
             pass
         return version
```

Comment:
Returning an empty string on exception is safe, but would it be more useful to return None or log a warning for SDK users who may rely on version info?

Reply:
@razorpay-sanjib Returning `None` would break the string formatting in User-Agent header. but we can add warning like this 

```
 warnings.warn(
    "Could not detect razorpay package version. Using fallback version. "
    "This may indicate an installation issue.",
    UserWarning,
     stacklevel=4
)
```